### PR TITLE
Add workflow for schema validation

### PR DESCRIPTION
**Pull request by marians**
### What this PR does / why we need it

This PR adds preliminary schema checks to this repository's GitHub actions, to ensure schema validity and avoid round trips. As we have a lot of movement in the schema currently, this is needed.

Note: the plan is to replace this simplistic version soon by one that is maintained automatically via devctl.

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade

### CODE REVIEW

**Comment by marians** [comment]
Currently fails because the schema is not completely normalized. Will be fixed via https://github.com/giantswarm/cluster-aws/pull/238

File: .github/workflows/json_schema.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          go-version: "^1.19"
+
+      - name: Install schemalint
+        run: |
+          go install github.com/giantswarm/schemalint@latest
+      - name: Run schemalint
+        run: |
+          schemalint verify ./helm/cluster-aws/values.schema.json --rule-set cluster-app
+
+  generate:
```
Comment (fiunchinho) [inline .github/workflows/json_schema.yaml:25]: `generate` sounds like we are generating something on this action, but we are really validating that the values were generated from the schema. I'd use a different name, but I'm bad at naming. maybe `generate-diff`? I don't know.

Comment (marians) [reply .github/workflows/json_schema.yaml:25]: True. I'm proposing `values-diff`.

**Review by fiunchinho — APPROVED** [review]

**Comment by marians** [comment]
@fiunchinho Can we merge this without e2e tests? (I think so)

**Comment by fiunchinho** [comment]
> @fiunchinho Can we merge this without e2e tests? (I think so)

yes

**Pull request merged**